apps/server/src/modules/tldraw/service/tldraw.ws.service.spec.ts

```diff
@@ -0,0 +1,376 @@
+		const testingModule = await Test.createTestingModule({
+			imports,
+			providers: [TldrawWs, TldrawBoardRepo, TldrawWsService],
+		}).compile();
+
+		service = testingModule.get<TldrawWsService>(TldrawWsService);
+		app = testingModule.createNestApplication();
+		app.useWebSocketAdapter(new WsAdapter(app));
+		jest.useFakeTimers({ advanceTimers: true, doNotFake: ['setInterval', 'clearInterval', 'setTimeout'] });
+		await app.init();
+	});
+
+	afterAll(async () => {
+		await app.close();
+	});
+
+	const createMessage = (values: number[]) => {
+		const encoder = encoding.createEncoder();
+		values.forEach((val) => {
+			encoding.writeVarUint(encoder, val);
+		});
+		encoding.writeVarUint(encoder, 0);
+		encoding.writeVarUint(encoder, 1);
+		const msg = encoding.toUint8Array(encoder);
+		return {
+			msg,
+		};
+	};
+
+	it('should service properties be defined', () => {
+		expect(service).toBeDefined();
+		expect(service.pingTimeout).toBeDefined();
+		expect(service.persistence).toBeDefined();
+	});
+
+	describe('when client is not connected to WS', () => {
+		const setup = async () => {
+			ws = await TestHelper.setupWs(wsUrl);
+
+			const closeConSpy = jest.spyOn(service, 'closeConn').mockImplementationOnce(() => {});
+			const sendSpy = jest.spyOn(service, 'send');
+			const doc: { conns: Map<WebSocket, Set<number>> } = { conns: new Map() };
+			const byteArray = new TextEncoder().encode(testMessage);
+
+			return {
+				closeConSpy,
+				sendSpy,
+				doc,
+				byteArray,
+			};
+		};
+
+		it('should throw error for send message', async () => {
+			const { closeConSpy, sendSpy, doc, byteArray } = await setup();
+
+			service.send(doc as WsSharedDocDo, ws, byteArray);
+
+			expect(sendSpy).toThrow();
+			expect(sendSpy).toHaveBeenCalledWith(doc, ws, byteArray);
+			expect(closeConSpy).toHaveBeenCalled();
+
+			ws.close();
+			sendSpy.mockRestore();
+		});
+	});
+
+	describe('when websocket has ready state different than 0 or 1', () => {
+		const setup = () => {
+			const closeConSpy = jest.spyOn(service, 'closeConn');
+			const sendSpy = jest.spyOn(service, 'send');
+			const doc: { conns: Map<WebSocket, Set<number>> } = { conns: new Map() };
+			const socketMock = { readyState: 3, close: () => {} };
+			const byteArray = new TextEncoder().encode(testMessage);
+
+			return {
+				closeConSpy,
+				sendSpy,
+				doc,
+				socketMock,
+				byteArray,
+			};
+		};
+
+		it('should close connection if websocket has ready state different than 0 or 1', () => {
+			const { closeConSpy, sendSpy, doc, socketMock, byteArray } = setup();
+
+			service.send(doc as WsSharedDocDo, socketMock as WebSocket, byteArray);
+
+			expect(sendSpy).toHaveBeenCalledWith(doc, socketMock, byteArray);
+			expect(sendSpy).toHaveBeenCalledTimes(1);
+			expect(closeConSpy).toHaveBeenCalled();
+
+			closeConSpy.mockRestore();
+			sendSpy.mockRestore();
+		});
+	});
+
+	describe('when websocket has ready state 0', () => {
```

Comment:
this one is outright confusing, this looks a lot like the description of the send test above, but now the send is actually mocked, and we are instead testing updateHandler..

Reply:
I've added describe('send') and assign this test to it (other test were also rearranged)